agent/connect/ca/provider_vault.go

```diff
@@ -737,6 +724,19 @@ func (v *VaultProvider) mountNamespaced(namespace, path string, mountInfo *vault
 	return err
 }
 
+func (v *VaultProvider) tuneMountNamespaced(namespace, path string, mountConfig *vaultapi.MountConfigInput) error {
```

Comment:
Could you explain what this does in place of the readNamespaced codepath it replaces?

Reply:
it's not a replacement for `readNamespaced` - it's basically the same as `mountNamespaced` above but for reconfiguring the pki mount for the intermediate CA cert. In the case where that mount already exists we have to call `/tune` instead.